compaction/compaction_manager.cc

```diff
@@ -600,7 +600,7 @@ void compaction_manager::submit(column_family* cf) {
                 task->compaction_running = false;
 
                 if (!can_proceed(task)) {
-                    maybe_stop_on_error(std::move(f), stop_iteration::yes);
```

Comment:
the reason why we had stop_iterator is because the caller can tell in advance if it wants a retry or not, like when compaction manager is being stopped.